File: backend/app/domain/entities/place.py
import uuid
from dataclasses import dataclass, field
from app.domain.value_objects.coordinates import Coordinates

@dataclass
class Factory:
    """Fábrica de origem das entregas, com coordenada geográfica."""
    name: str
    location: Coordinates
    id: uuid.UUID = field(default_factory=uuid.uuid4)

@dataclass
class Store:
    """Loja de destino das entregas, vinculada a um lojista (owner_id)."""
    name: str
    location: Coordinates
    owner_id: uuid.UUID
    id: uuid.UUID = field(default_factory=uuid.uuid4)

File: frontend/src/domain/entities/Place.ts
import { Entity } from './Entity';
import { Coordinates } from '../value_objects/Coordinates';

export interface FactoryProps {
  name: string;
  location: Coordinates;
}

export class Factory extends Entity<FactoryProps> {
  get name(): string { return this.props.name; }
  get location(): Coordinates { return this.props.location; }
}

export interface StoreProps {
  name: string;
  location: Coordinates;
  ownerId: string;
}

export class Store extends Entity<StoreProps> {
  get name(): string { return this.props.name; }
  get location(): Coordinates { return this.props.location; }
  get ownerId(): string { return this.props.ownerId; }
}
